cmd/flow-capture.go

```diff
@@ -0,0 +1,192 @@
+package cmd
+
+import (
+	"bufio"
+	"encoding/json"
+	"net"
+	"net/textproto"
+	"os"
+	"os/exec"
+	"sort"
+	"time"
+
+	"github.com/jpillora/sizestr"
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	"github.com/spf13/cobra"
+
+	"github.com/fatih/color"
+	"github.com/rodaine/table"
+)
+
+const (
+	flowsToShow = 40
+)
+
+var flowCmd = &cobra.Command{
+	Use:   "get-flows",
+	Short: "",
+	Long:  "",
+	Run:   runFlowCapture,
+}
+
+var lastFlows = []config.GenericMap{}
+
+func runFlowCapture(cmd *cobra.Command, args []string) {
+	wg.Add(len(addresses))
+	for i, _ := range addresses {
+		go runFlowCaptureOnAddr(addresses[i], nodes[i])
+	}
+	wg.Wait()
+}
+
+func runFlowCaptureOnAddr(addr string, filename string) {
+	log.Infof("Starting Flow Capture for %s...", filename)
+
+	tcpServer, err := net.ResolveTCPAddr("tcp", addr)
+
+	if err != nil {
+		log.Error("ResolveTCPAddr failed:", err.Error())
+		log.Fatal(err)
+	}
+	conn, err := net.DialTCP("tcp", nil, tcpServer)
+	if err != nil {
+		log.Error("Dial failed:", err.Error())
+		log.Fatal(err)
+	}
+	reader := bufio.NewReader(conn)
+	tp := textproto.NewReader(reader)
+	var f *os.File
+	err = os.MkdirAll("./output/flow/", 0700)
+	if err != nil {
+		log.Errorf("Create directory failed: %v", err.Error())
+		log.Fatal(err)
+	}
+	f, err = os.Create("./output/flow/" + filename)
+	if err != nil {
+		log.Errorf("Create file %s failed: %v", filename, err.Error())
+		log.Fatal(err)
+	}
```

Comment:
It would be very useful if you can choose the stdout format, which could just be JSON instead of the UI table.  I know technically you can effectively get this with the 'tee' command, but if it didn't output to the UI table, it would be more scalable.  This would be simpler to export flows than what we have today.

Reply:
I've added some predefined view capabilities, including raw JSON :+1: